autofit/optimize/non_linear/nested_sampling/multi_nest.py

```diff
@@ -210,279 +209,148 @@ def __call__(self, cube, ndim, nparams, lnew):
         )
         self.paths.backup()
 
-        instance = multinest_output.most_likely_instance
-        multinest_output.output_results(during_analysis=False)
+        samples = self.samples_from_model(model=model, paths=self.paths)
+
+        instance = samples.max_log_likelihood_instance
+        samples.output_results(during_analysis=False)
         return Result(
             instance=instance,
-            likelihood=multinest_output.maximum_log_likelihood,
-            output=multinest_output,
+            log_likelihood=samples.max_log_posterior,
+            samples=samples,
             previous_model=model,
-            gaussian_tuples=multinest_output.gaussian_priors_at_sigma(self.sigma),
+            gaussian_tuples=samples.gaussian_priors_at_sigma(self.sigma),
         )
 
-    def output_from_model(self, model, paths):
-        """Create this non-linear search's output class from the model and paths.
+    def samples_from_model(self, model: AbstractPriorModel, paths):
+        """Create a *Samples* object from this non-linear search's output files on the hard-disk and model.
 
-        This function is required by the aggregator, so it knows which output class to generate an instance of."""
-        return MultiNestOutput(model=model, paths=paths)
+        For MulitNest, this requires us to load:
 
+            - The parameter samples, log likelihood values and weights from the multinest.txt file.
+            - The total number of samples (e.g. accepted + rejected) from resume.dat.
+            - The log evidence of the model-fit from the multinestsummary.txt file (if this is not yet estimated a
+              value of -1.0e99 is used.
 
-class MultiNestOutput(NestedSamplerOutput):
-    @property
-    def pdf(self):
-        """An interface to *GetDist* which can be used for analysing and visualizing the non-linear search chains.
+        Parameters
+        ----------
+        model
+            The model which generates instances for different points in parameter space. This maps the points from unit
+            cube values to physical values via the priors.
+        paths : af.Paths
+            A class that manages all paths, e.g. where the phase outputs are stored, the non-linear search chains,
+            backups, etc.
+        """
 
-        *GetDist* can only be used when chains are converged enough to provide a smooth PDF and this convergence is
-        checked using the *pdf_converged* bool before *GetDist* is called.
+        parameters = parameters_from_file_weighted_samples(file_weighted_samples=paths.file_weighted_samples,
+                                                           prior_count=model.prior_count)
+        log_priors = [sum(model.log_priors_from_vector(vector=vector)) for vector in parameters]
+        log_likelihoods = log_likelihoods_from_file_weighted_samples(file_weighted_samples=paths.file_weighted_samples)
```

Comment:
This dense block of text would be a lot more readable if arguments were distributed across new lines